### DISCUSSION TITLE
Source Generators - Unity3d

### DISCUSSION THREAD

**Discussion by jjolley** [dotnet/roslyn#46124] [General]
Will we be able to use source generators with Unity3d?

The performance benefits could be substantial.

**Comment by sharwell** [comment]
Source generators should be usable in any situation where all of the following hold:

1. The project can use C# 9 or newer (though it's possible this feature will remain under the "preview" language version - the compiler team can provide a more definitive answer).
2. The project can use analyzers. Source generators use the same API mechanism that analyzers use, so if the project can use analyzers then it should be able to use source generators.

**Comment by jjolley** [comment]
Great! Thanks for letting me know. Very excited for this feature.
